src/navsat_transform.cpp

```diff
@@ -51,55 +51,63 @@
 
 using std::placeholders::_1;
 using std::placeholders::_2;
+using namespace std::chrono_literals;
 
 namespace robot_localization
 {
-NavSatTransform::NavSatTransform()
-: magnetic_declination_(0.0), utm_odom_tf_yaw_(0.0), yaw_offset_(0.0),
-  transform_timeout_(0.0), broadcast_utm_transform_(false),
+NavSatTransform::NavSatTransform(const rclcpp::NodeOptions & options) : 
+  Node("navsat_transform_node", options),
+  base_link_frame_id_("base_link"),
+  broadcast_utm_transform_(false),
   broadcast_utm_transform_as_parent_frame_(false),
-  has_transform_odom_(false), has_transform_gps_(false),
-  has_transform_imu_(false), transform_good_(false), gps_frame_id_(""),
-  gps_updated_(false), odom_updated_(false), publish_gps_(false),
-  use_odometry_yaw_(false), use_manual_datum_(false), zero_altitude_(false),
-  world_frame_id_("odom"), base_link_frame_id_("base_link"), utm_zone_(""),
-  tf_listener_(tf_buffer_)
+  gps_frame_id_(""),
+  gps_updated_(false),
+  has_transform_gps_(false),
+  has_transform_imu_(false),
+  has_transform_odom_(false),
+  magnetic_declination_(0.0),
+  odom_updated_(false),
+  publish_gps_(false),
+  tf_buffer_(this->get_clock()),
+  tf_listener_(tf_buffer_),
+  transform_good_(false), 
+  transform_timeout_(0ns),
+  use_manual_datum_(false), 
+  use_odometry_yaw_(false), 
+  utm_broadcaster_(*this),
+  utm_odom_tf_yaw_(0.0), 
+  utm_zone_(""), 
+  world_frame_id_("odom"),
+  yaw_offset_(0.0),
+  zero_altitude_(false)
 {
-  node_ = rclcpp::node::Node::make_shared("navsat_transform_node");
-
   latest_utm_covariance_.resize(POSE_SIZE, POSE_SIZE);
   latest_odom_covariance_.resize(POSE_SIZE, POSE_SIZE);
-}
 
-NavSatTransform::~NavSatTransform() {}
-
-void NavSatTransform::run()
-{
   double frequency = 10.0;
   double delay = 0.0;
   double transform_timeout = 0.0;
 
   // Load the parameters we need
-  magnetic_declination_ = node_->get_parameter("magnetic_declination_radians");
-  node_->get_parameter_or("yaw_offset", yaw_offset_, 0.0);
-  node_->get_parameter_or("broadcast_utm_transform", broadcast_utm_transform_,
-    false);
-  node_->get_parameter_or("broadcast_utm_transform_as_parent_frame",
-    broadcast_utm_transform_as_parent_frame_, false);
-  node_->get_parameter_or("zero_altitude", zero_altitude_, false);
-  node_->get_parameter_or("publish_filtered_gps", publish_gps_, false);
-  node_->get_parameter_or("use_odometry_yaw", use_odometry_yaw_, false);
-  node_->get_parameter_or("wait_for_datum", use_manual_datum_, false);
-  node_->get_parameter_or("frequency", frequency, 10.0);
-  node_->get_parameter_or("delay", delay, 0.0);
-  node_->get_parameter_or("transform_timeout", transform_timeout, 0.0);
+  magnetic_declination_ = this->declare_parameter("magnetic_declination_radians", 0.0);
```

Comment:
So parameters are one place I am _really_ out of the loop for ROS2. I added the ROS2 feature to support parameter arrays, but lost touch with it after that. I know ROS2 handles parameters at the node level, so I'm guessing this is the canonical way to define what parameters a node supports.

Reply:
Yeah, so you have to declare variables and their types (or implied by defaults) for them to be get/set-able.